homeassistant/components/cloud/backup.py

```diff
@@ -170,6 +166,51 @@ async def async_upload_backup(
         except (TimeoutError, ClientError) as err:
             raise BackupAgentError("Failed to upload backup") from err
 
+    async def async_upload_backup(
+        self,
+        *,
+        open_stream: Callable[[], Coroutine[Any, Any, AsyncIterator[bytes]]],
+        backup: AgentBackup,
+        **kwargs: Any,
+    ) -> None:
+        """Upload a backup.
+
+        :param open_stream: A function returning an async iterator that yields bytes.
+        :param backup: Metadata about the backup that should be uploaded.
+        """
+        if not backup.protected:
+            raise BackupAgentError("Cloud backups must be protected")
+
+        base64md5hash = await _b64md5(await open_stream())
+        filename = self._get_backup_filename()
+        metadata = backup.as_dict()
+        size = backup.size
+
+        tries = 1
+        while tries <= _RETRY_LIMIT:
+            try:
+                await self._async_do_upload_backup(
+                    open_stream=open_stream,
+                    filename=filename,
+                    base64md5hash=base64md5hash,
+                    metadata=metadata,
+                    size=size,
+                )
+                break
+            except BackupAgentError as err:
+                if tries == _RETRY_LIMIT:
+                    raise
+                tries += 1
+                retry_timer = random.randint(60, 600)
+                _LOGGER.debug(
```

Comment:
I think we should increase verbosity here to info
```suggestion
                _LOGGER.info(
```